src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1110,231 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
+         if (field_name != m_meshNodesGFName)
+         {
+            SLIC_WARNING("QField with the name '" << field_name<< "' is already "
+                         "registered, overwriting the old qfield");
+         }
+#endif
+         DeregisterQField(field_name);
+      }
+   }
+
+   sidre::Group* grp = f->createGroup( field_name );
+
+   // Set the "qspace_order" string using the qf's qspace, overwrite if
+   // necessary.
+   sidre::View *v = alloc_view(grp, "qspace_order");
+   v->setScalar(qf->GetSpace()->GetOrder());
+
+   // Set the "vdim" string using the qf's qspace, overwrite if
+   // necessary.
+   v = alloc_view(grp, "vdim");
+   v->setScalar(qf->GetVDim());
+
+   // Set the topology of the QuadratureFunction.
+   // This is always 'mesh'
+   v = alloc_view(grp, "topology")->setString("mesh");
+
+   // Set the View "<m_bp_grp>/qfields/<field_name>/values"
+   addQuadratureFunction(field_name, qf, buffer_name, offset);
+
+
+   // Register field_name in the blueprint_index group.
+   if (myid == 0)
+   {
+      RegisterQFieldInBPIndex(field_name, qf);
+   }
+
+   // Register field_name + qf in field_map.
+   mfem::DataCollection::RegisterQField(field_name, qf);
+}
+
+// private method
+// Should only be called on mpi rank 0 ( or if serial problem ).
+void MFEMSidreDataCollection::
+DeregisterQFieldInBPIndex(const std::string& field_name)
+{
+   sidre::Group * fields_grp = m_bp_index_grp->getGroup("qfields/");
+   SLIC_WARNING_IF(fields_grp->hasGroup(field_name),
+               "No qfield exists in blueprint index with name " << field_name);
+
+   // Note: This will destroy all orphaned views or buffer classes under this
+   // group also.  If sidre owns this field data, the memory will be deleted
+   // unless it's referenced somewhere else in sidre.
+   fields_grp->destroyGroup(field_name);
+}
+
+void MFEMSidreDataCollection::DeregisterQField(const std::string& field_name)
+{
+   // Deregister field_name from field_map.
+   mfem::DataCollection::DeregisterQField(field_name);
+
+   sidre::Group * fields_grp = m_bp_grp->getGroup("qfields/");
+   SLIC_WARNING_IF(fields_grp->hasGroup(field_name),
+               "No qfield exists in blueprint with name " << field_name);
+
+   // Delete field_name from the blueprint group.
+
+   // Note: This will destroy all orphaned views or buffer classes under this
+   // group also.  If sidre owns this field data, the memory will be deleted
+   // unless it's referenced somewhere else in sidre.
+   fields_grp->destroyGroup(field_name);
+
+   // Delete field_name from the blueprint_index group.
+   if (myid == 0)
+   {
+      DeregisterQFieldInBPIndex(field_name);
+   }
+
+   // Delete field_name from the named_buffers group, if allocated.
+   FreeNamedBuffer(field_name);
+}
+
+int MFEMSidreDataCollection::GetQFieldOrder(const std::string &field_name)
+{
+   sidre::Group* grp = m_bp_grp->getGroup("qfields/" + field_name);
+   SLIC_ASSERT_MSG(grp != NULL, 
```

Comment:
Note for the future: We should look into how error checking / reporting is done in this class and make sure we are consistent about runtime check vs. ASSERT (debug-only) checks.